### PR TITLE
docs(authentication.md): add authentication doc

### PR DESCRIPTION
**Pull request by vdice**
* Adds authentication doc

Preview: https://deploy-preview-1571--brigade-docs.netlify.app/topics/administrators/authentication/

Closes https://github.com/brigadecore/brigade/issues/1570

### CODE REVIEW

File: docs/content/topics/administrators/authentication.md [hunk 1]
```diff
@@ -11,4 +11,77 @@ aliases:
 
 # Brigade Authentication
 
-TODO
+Brigade utilizes third-party providers for its user authentication strategy.
+We'll go over the available options and their configuration in this document.
+
+Note that Brigade's default mode of root user access precludes functionality
+related to user management or authorization. Therefore, selection of a viable
+third-party auth strategy (and thus disabling root user mode) is a
```
Comment (vdice) [inline docs/content/topics/administrators/authentication.md:19]: Actually, looking for thoughts from @krancour here.  I had always assumed a binary root user mode OR third-party auth, but there isn't anything in the Brigade chart preventing deployment of both modes enabled.  If that's how we'd like it to continue to be, I can remove the `(and thus disabling root user mode)` as that wouldn't strictly be necessary to use third-party auth.

File: docs/content/topics/administrators/authentication.md [hunk 2]
```diff
@@ -11,4 +11,77 @@ aliases:
...
+Note: since these privileges are granted on first login, adding/revoking these
+permissions for users who have already logged in for the first time must be
+done via the brig CLI directly rather than re-deploying with differing
+configuration.
+
+[values.yaml]: https://github.com/brigadecore/brigade/blob/v2/charts/brigade/values.yaml
+
+### GitHub OAuth Provider
+
+To use [GitHub's Oauth Provider], please see the
```
Comment (vdice) [inline docs/content/topics/administrators/authentication.md:83]: A few other options here; I could go either way.  We could duplicate the GitHub OAuth setup instructions here (but then have two areas to maintain/update) or we could put the full instructions here and move the link to the Deploy doc.  Or, of course, leave as is.

File: docs/content/topics/administrators/authentication.md (hunk 1 above)
Comment (krancour) [reply docs/content/topics/administrators/authentication.md:19]: You can do both. It's like Linux. root is a thing, but you shouldn't share it-- which is where the 3rd party auth comes in.

File: docs/content/topics/administrators/authentication.md [hunk 3]
```diff
@@ -11,4 +11,77 @@ aliases:
 
 # Brigade Authentication
 
-TODO
+Brigade utilizes third-party providers for its user authentication strategy.
```
Comment (krancour) [inline docs/content/topics/administrators/authentication.md:14]: Do you think it's worthwhile to explain the rationale for this? I could go either way.

The rationale is that with Brigade 2 providing an abstraction between users and the underlying substrate (Kubernetes), it cannot be taken for granted that every user of Brigade is also a Kubernetes user. That introduces a need for sourcing user identity from somewhere else.

File: docs/content/topics/administrators/authentication.md (hunk 2 above)
Comment (krancour) [reply docs/content/topics/administrators/authentication.md:83]: imho, the best option is to move those instructions to here and link to it from the deploy doc.

**Comment by krancour** [comment]
Overall, looks good.

File: docs/content/topics/administrators/authentication.md (hunk 3 above)
Comment (vdice) [reply docs/content/topics/administrators/authentication.md:14]: Good call, I'd been taking that point for granted, definitely worth a mention.  Will add.

Comment (krancour) [reply docs/content/topics/administrators/authentication.md:14]: My wording was quick and dirty tho. I'm sure you can improve on it.

Comment (vdice) [reply docs/content/topics/administrators/authentication.md:14]: Ok, updated.  PTAL when convenient.  Thanks!

File: docs/content/topics/administrators/authentication.md [hunk 4]
```diff
@@ -11,4 +11,154 @@ aliases:
...
+enabling the suite of user authorization commands.
+## Background
+
+The motivation behind this design stems from one of the core guiding principles
+in Brigade 2: Provide an abstraction between users and the underlying substrate
+(Kubernetes), such that users of Brigade do not need to interact with (or have
+any presence on) the substrate itself. Therefore, the need to source user
+identity from another authority became apparent. Rather than implementing such
+a system in Brigade itself, we've decided to utilize well-known external
+services to fulfill this need.
```
Comment (krancour) [inline docs/content/topics/administrators/authentication.md:27]: ❤️   Great explanation.

**Review by krancour — APPROVED** [review]
LGTM!

**Comment by vdice** [comment]
Apologies, @krancour , I noticed an opportunity for a few more tweaks after your LGTM.  Please see https://github.com/brigadecore/brigade/pull/1571/commits/0cd3a29b1172ac94a34a53a4a04c08a9b8bb6c3a

**Comment by krancour** [comment]
@vdice. No apologies necessary. Still LGTM.

**Pull request merged**